optimade/adapters/structures/adapter.py

```diff
@@ -55,4 +57,10 @@ class Structure(EntryAdapter):
 
     _type_ingesters: Dict[str, Callable] = {
         "pymatgen": from_pymatgen,
+        "ase": from_ase_atoms,
+    }
+
+    _type_ingesters_by_type: Dict[str, Type] = {
+        "pymatgen": PymatgenStructure,
+        "ase": ASEAtoms,
```

Comment:
if you accepted my previous comment, this becomes
```suggestion
         PymatgenStructure: "pymatgen",
         ASEAtoms: "ase",
```

Reply:
I thought about doing this, perhaps it is a bit neater. I'll try it out and push the changes

Reply:
Hmm, actually, I think `isinstance` is more robust than `type` here, as it can handle inheritance, e.g.,

```python
>>> type(Structure(lattice=[[1,1,1],[1,1,1],[1,1,1]], species=[], coords=[])) == IStructure
False
```

having the values be types also makes it easier to extend to multiple types in the future, e.g.,

```
{
        "pymatgen": (PymatgenStructure, IStructure)
        "ase": ASEAtoms,
}
```

Reply:
Yes, that is a good reason to keep things the way they are.